addons/sale/wizard/sale_payment_register.py

```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models
+
+
+class SalePaymentRegister(models.TransientModel):
+    _inherit = 'account.payment.register'
+
+    def _create_payment_vals_from_wizard(self):
+        vals = super(SalePaymentRegister, self)._create_payment_vals_from_wizard()
+        vals.update({'team_id': self.line_ids[0].move_id.team_id.id})
```

Comment:
```suggestion
        vals = super()._create_payment_vals_from_wizard()
        # Make sure the account move linked to generated payment
        # belongs to the expected sales team
        vals.update({'team_id': self.line_ids.move_id[0].team_id.id})
```

Had to investigate a bit before finding where this team_id field is specified, bc it's coming from an `_inherits`.
So we might as well precise this info here directly to be clearer :).

Also, no need to specify class in super call, unless you want to modify self (`super(Class, self.with_context(x=y)`)

Also', I moved the `[0]` part bc I believe it makes more sense conceptually :thinking:, even if it shouldn't change anything in the final result.